README.md

```diff
@@ -1,2 +1,1584 @@
+it retrieves these images from the registries and then stores them locally.
+
+The device Supervisor then looks at the images (and any docker-compose manifest
+associated with them) to determine how to create those services as containers,
+which form executable instances of an image.
+
+Additionally, volumes (persistent data stores) can be bound to an executble
+container to ensure that data exists across instances of the images.
+
+## 1.1 Service Images
+
+Service images are the basis for any service. They comprise the binaries and
+data for running services, but they also contain metadata that informs
+balenaEngine on how to initialise and configure the service. As such, an image
+is essentially a self-enclosed Linux filesystem with everything required for
+running an instance of Linux, except for the kernel.
+
+Images are produced by the balena builders, which use Dockerfiles to determine
+what to include in the image. See [here](https://docs.docker.com/engine/reference/builder/)
+for more information on Dockerfiles.
+
+## 1.2 Service Containers
+
+When a service container is created, balenaEngine uses a service image as a
+'template' for that container, containing all of the files that the image did
+as well as using the metadata associated with it for configuration. It then
+creates local filesystem for that service container to .
+
+As such, service containers are writable allowing data to be written and read
+from any directory within that service on these locally created filesystem
+layers. However, these layers only exist for the lifespan of the container.
+Containers are recreated for multiple reasons. These can include newer versions
+of a service image being downloaded, alterations to environment variables names
+or values, a change in network configuration, etc.
+
+As such, it is never safe to rely on data existing in a service container layer
+above that of the lifespan of that service container. Temporary execution data
+is therefore acceptable, but any data that is required to persist across
+container recreation or restarts should be stored in persistent volumes.
+
+Note that service recreation is *not* the same as restarting. Restarting a
+service does not require it's recreation, and implies that nothing has changed
+in the configuration requirements of the service container. However, it can
+be tricky to determine when a service might be recreated or restarted.
+
+## 1.3 Service Volumes
+
+Volumes are persistent storage areas which do not get erased when a service
+container is recreated. These volumes are usually bound to a service container
+on a specific path, and are stored on the host filesystem without any
+sort of diff mechanism (that is, any file hierarchy data written to the volume
+is always available in that hierarchy).
+
+For single service applications, there is a default persistent volume bound to
+the `/data` directory in the service. This path can always be considered
+persistent.
```

Comment:
in the single container case only. we have had people expect it to also to be there in multicontainer and been confused when data is not persisted.